library-best-practices/JLBP-2.md

```diff
@@ -2,19 +2,40 @@
 -----------------------------
 
 - Avoid exposing types from your dependencies.
-  - Your dependencies have their own rules about how often
-    they do major version bumps and make breaking changes.
-    Their frequency of such changes may be more often than yours,
-    which causes problems if you want to retain a stable surface for
-    longer than your dependencies. Historically this was a
-    big problem for Guava, which made a major version bump
-    every 6 months, although Guava no longer does this.
-- Use package-protected classes and methods when the API is internal
-  only and should not be used by consumers.
-- Do not mark methods and classes public by default.
-  Assume non-public until a need is known.
+
+  - Method arguments and return values from public methods should be standard Java 
+    types such as `java.time.LocalDate` or classes defined in the library itself,
+    not third party types such as `org.joda.time.LocalDate`.
+
+  - Your own public types should not be subclasses or implementations of types
+    in third party libraries.  
+
+  - Third party libraries can change their own API more frequently than you like
+    or do so at inconvenient times. If a third party type on the surface of
+    your API changes or is removed, you either have to break your own API or remain with an older,
+    unsupported version of the library that is likely to cause diamond dependency
+    problems. Historically this was a big problem for libraries that exposed Guava types
+    such as `com.google.common.io.OutputSupplier` because Guava bumped major versions
```

Comment:
bump -> incremented (to use specific verb)

Reply:
done